bcf_sr_sort.c

```diff
@@ -546,23 +558,8 @@ static void bcf_sr_sort_set(bcf_srs_t *readers, sr_sort_t *srt, const char *chr,
 int bcf_sr_sort_next(bcf_srs_t *readers, sr_sort_t *srt, const char *chr, int min_pos)
 {
     int i,j;
-    if ( readers->nreaders == 1 )
-    {
-        srt->nsr = 1;
-        if ( !srt->msr )
-        {
-            srt->vcf_buf = (vcf_buf_t*) calloc(1,sizeof(vcf_buf_t));   // first time here
-            srt->msr = 1;
-        }
-        bcf_sr_t *reader = &readers->readers[0];
-        assert( reader->buffer[1]->pos==min_pos );
-        bcf1_t *tmp = reader->buffer[0];
-        for (j=1; j<=reader->nbuffer; j++) reader->buffer[j-1] = reader->buffer[j];
-        reader->buffer[ reader->nbuffer ] = tmp;
-        reader->nbuffer--;
-        readers->has_line[0] = 1;
-        return 1;
-    }
+    assert( srt->nactive>0 );
```

Comment:
Is this assert something that is there for protection against coding errors? Ie absolutely regardless of user input, well formed or otherwise, ```src->nactive``` will always be positive.  Or is it something that is protection for malformed data files?

If the former then it is fine, but if the latter then it should be replaced by an appropriate return code to signify error and handled by the calling application properly.

I don't know enough about this and the other asserts to be sure.

Reply:
This is only a protection against coding errors.

Reply:
Thanks.  No issue with that then.